specs/clipboard-api/clipboard.idl

```diff
@@ -0,0 +1,426 @@
+  /// </summary>
+  [contract(ClipboardContract, 1.0)]
+  runtimeclass DataPackagePropertySet :
+    [default] IDataPackagePropertySet
+    , Windows.Foundation.Collections.IMap<String, Object>
+    , Windows.Foundation.Collections.IIterable<Windows.Foundation.Collections.IKeyValuePair<String, Object> >
+  { }
+
+  [contract(ClipboardContract, 1.0)]
+  [exclusiveto(DataPackagePropertySet)]
+  [hasvariant]
+  interface IDataPackagePropertySet requires
+    Windows.Foundation.Collections.IMap<String, Object>
+    , Windows.Foundation.Collections.IIterable<Windows.Foundation.Collections.IKeyValuePair<String, Object> >
+  {
+    String Title;
+    String Description;
+    Windows.Storage.Streams.IRandomAccessStreamReference Thumbnail;
+    Windows.Foundation.Collections.IVector<String> FileTypes { get; };
+    String ApplicationName;
+    Windows.Foundation.Uri ApplicationListingUri;
+
+    Windows.Foundation.Uri ContentSourceWebLink;
+    Windows.Foundation.Uri ContentSourceApplicationLink;
+    String PackageFamilyName;
+    Windows.Storage.Streams.IRandomAccessStreamReference Square30x30Logo;
+    Windows.UI.Color LogoBackgroundColor;
+
+    String EnterpriseId;
+
+    String ContentSourceUserActivityJson;
+  }
+
+  /// <summary>
+  /// Read-only view of a <see cref="DataPackagePropertySet"/>.
+  /// </summary>
+  [contract(ClipboardContract, 1.0)]
+  runtimeclass DataPackagePropertySetView :
+    [default] IDataPackagePropertySetView
+    , Windows.Foundation.Collections.IMapView<String, Object>
+    , Windows.Foundation.Collections.IIterable<Windows.Foundation.Collections.IKeyValuePair<String, Object> >
+  { }
+
+  [contract(ClipboardContract, 1.0)]
+  [exclusiveto(DataPackagePropertySetView)]
+  [hasvariant]
+  interface IDataPackagePropertySetView requires
+    Windows.Foundation.Collections.IMapView<String, Object>
+    , Windows.Foundation.Collections.IIterable<Windows.Foundation.Collections.IKeyValuePair<String, Object> >
+  {
+    String Title { get; };
+    String Description { get; };
+    Windows.Storage.Streams.IRandomAccessStreamReference Thumbnail { get; };
+    Windows.Foundation.Collections.IVectorView<String> FileTypes { get; };
+    String ApplicationName { get; };
+    Windows.Foundation.Uri ApplicationListingUri { get; };
+
+    String PackageFamilyName { get; };
+    Windows.Foundation.Uri ContentSourceWebLink { get; };
+    Windows.Foundation.Uri ContentSourceApplicationLink { get; };
+    Windows.Storage.Streams.IRandomAccessStreamReference Square30x30Logo { get; };
+    Windows.UI.Color LogoBackgroundColor { get; };
+
+    String EnterpriseId { get; };
+
+    String ContentSourceUserActivityJson { get; };
+
+    Boolean IsFromRoamingClipboard { get; };
+  }
+
+  /// <summary>
+  /// An acknowledgement by the system that your app is in the process of
+  /// a delay rendering, with the ability to signal the system when done.
+  /// </summary>
+  [contract(ClipboardContract, 1.0)]
+  runtimeclass DataProviderDeferral

```

Comment:
I’m really not a fan of these “deferral” types. More specifically, when I am writing an AppContainer app, I may need time to ensure that all of my unsaved data has been flushed to persistent storage. However, even if I take a deferral, the system is still able to interrupt my saving code, potentially causing data corruption or loss. Worse yet, the documentation does not definitively state under what circumstances my deferral will be ignored (it’s “up to Windows” if this happens or not). I’d appreciate it if we could take then into mind when documenting the API surface.

Reply:
Hm, that sounds more like a general problem with the BlahDeferral concept than specific to clipboard or app-to-app data transfer. If you don't mind, could you fork that into a separate issue or discussion?